kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -752,8 +780,32 @@ private void generateClusteringDomainMetadataIfNeeded() {
           && clusteringColumnsOpt.isPresent()) {
         DomainMetadata clusteringDomainMetadata =
             ClusteringUtils.getClusteringDomainMetadata(clusteringColumnsOpt.get());
-        domainsToAdd.put(clusteringDomainMetadata.getDomain(), clusteringDomainMetadata);
+        addDomain(
+            clusteringDomainMetadata.getDomain(), clusteringDomainMetadata.getConfiguration());
       }
     }
   }
+
+  /**
+   * Returns the remove file rows to completely reset the table state in the case of a replace
```

Comment:
`to completely reset the table state in the case of a replace (i.e. removes every active add file in the table)` 

This seems like 2 different things to me. The former seems like a superset of the latter. i.e. the former would also include resetting the schema? / protocol?

Can you just say: ` Returns the remove file rows needed to remove every active add file in the table`?

Reply:
Makes sense sgtm